homeassistant/components/aladdin_connect/config_flow.py

```diff
@@ -1,137 +1,58 @@
-"""Config flow for Aladdin Connect cover integration."""
-
-from __future__ import annotations
+"""Config flow for Aladdin Connect Genie."""
 
 from collections.abc import Mapping
+import logging
 from typing import Any
 
-from AIOAladdinConnect import AladdinConnectClient
-import AIOAladdinConnect.session_manager as Aladdin
-from aiohttp.client_exceptions import ClientError
 import voluptuous as vol
 
-from homeassistant.config_entries import ConfigEntry, ConfigFlow, ConfigFlowResult
-from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
-from homeassistant.core import HomeAssistant
-from homeassistant.exceptions import HomeAssistantError
-from homeassistant.helpers.aiohttp_client import async_get_clientsession
-
-from .const import CLIENT_ID, DOMAIN
-
-STEP_USER_DATA_SCHEMA = vol.Schema(
-    {
-        vol.Required(CONF_USERNAME): str,
-        vol.Required(CONF_PASSWORD): str,
-    }
-)
-
-REAUTH_SCHEMA = vol.Schema({vol.Required(CONF_PASSWORD): str})
-
-
-async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> None:
-    """Validate the user input allows us to connect.
+from homeassistant.config_entries import ConfigEntry, ConfigFlowResult
+from homeassistant.helpers import config_entry_oauth2_flow
 
-    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
-    """
-    acc = AladdinConnectClient(
-        data[CONF_USERNAME],
-        data[CONF_PASSWORD],
-        async_get_clientsession(hass),
-        CLIENT_ID,
-    )
-    try:
-        await acc.login()
-    except (ClientError, TimeoutError, Aladdin.ConnectionError):
-        raise
+from .const import CONFIG_FLOW_MINOR_VERSION, CONFIG_FLOW_VERSION, DOMAIN
 
-    except Aladdin.InvalidPasswordError as ex:
-        raise InvalidAuth from ex
 
+class OAuth2FlowHandler(
```

Comment:
One major thing I am missing is, can we set some unique user id as unique_id for the config flow? This now creates the situation where someone can break their installation by logging in with the wrong account. We should prevent that.